src/artm/core/collection_parser.cc

```diff
@@ -563,15 +590,92 @@ CollectionParserInfo CollectionParser::ParseVowpalWabbit() {
 
           transaction_types.emplace(TransactionType(class_ids));
           batch_collector.Record(class_ids, tokens, transaction_weight);
-        }
 
+          if (config.gather_cooc()) {
+            if (transaction_types.size() > 1) {
```

Comment:
this is not what I meant(

transaction_types elements are tuples like <class_id1, class_id2, ...>, in case of regular non-transaction models each element contains only one class_id. So you need to check this fact. Now you're checking number of transaction types, e.g. number of class_ids, that is > 1 in all multimodal models